app/assets/javascripts/components/item_form.es6.jsx

```diff
@@ -0,0 +1,119 @@
+class ItemForm extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      user_id: props.userId,
+      list_id: props.listId,
+      name: '',
+      purchased: false,
+      quantity: '',
+      quantity_name: ''
+    }
+  }
+
+  handleUserInput(event) {
+    const name = event.target.name;
+    const obj = {};
+    obj[name] = event.target.value;
+    this.setState(obj);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const item = {
+      user_id: this.state.user_id,
+      name: this.state.name,
+      list_id: this.state.list_id,
+      quantity: this.state.quantity,
+      purchased: this.state.purchased,
+      quantity_name: this.state.quantity_name
+    }
+    $.post('/items', { item: item }).done((data) => {
+      this.props.handleItemAddition(data);
+    });
+    this.setState({
+      name: '',
+      purchased: false,
+      quantity: '',
+      quantity_name: ''
+    });
+  }
+
+  // Typeahead stuff will need to be re-engineered once
+  // upgrading to React is complete
```

Comment:
The reason for this is because until we update to at least `react_on_rails` we aren't using yarn or npm and access to JS libraries is kind of jacked up.